packages/injectable/ogre-tools-injectable.d.ts

```diff
@@ -101,25 +101,21 @@ declare module '@ogre-tools/injectable' {
     ): AsyncReturnable<TReturnAsPromise, Instance[]>;
   }
 
-  export interface ILifecycle<InstantiationParam = void, Key extends string | number | symbol = string | number | symbol> {
+  export interface ILifecycle<InstantiationParam, Key extends string | number | symbol> {
     getInstanceKey: (di: DiContainer, params: InstantiationParam) => Key;
   }
 
-  export interface KeyedSingletonOptions<InstantiationParam = void> {
-    getInstanceKey: (di: DiContainer, params: InstantiationParam) => string | number | symbol;
-  }
-
-  export type KeyedSingleton = <InstantiationParam = void>(keyedSingletonOptions: KeyedSingletonOptions<InstantiationParam>) => ILifecycle<InstantiationParam>;
-
   const storedInstanceKey: unique symbol;
   const nonStoredInstanceKey: unique symbol;
 
   export const lifecycleEnum: {
-    singleton: ILifecycle<[], typeof storedInstanceKey>;
+    singleton: ILifecycle<void, typeof storedInstanceKey>;
 
-    keyedSingleton: KeyedSingleton;
+    keyedSingleton<InstantiationParam>(
+      options: ILifecycle<InstantiationParam, string | number | symbol>,
```

Comment:
Idea: the underlying code uses js-Map, therefore we don't really need to limit the type of instance keys to eg. `string | number | symbol`.

Reply:
So just `any`?